test/integration/node-specific/mongo_client.test.ts

```diff
@@ -0,0 +1,605 @@
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+      }
+    );
+  });
+
+  context('implicit #connect()', () => {
+    let client: MongoClient;
+    beforeEach(function () {
+      client = this.configuration.newClient(this.configuration.url(), {
+        monitorCommands: true
+      });
+    });
+
+    afterEach(async function () {
+      await client.close();
+    });
+
+    it(
+      'automatically connects upon first operation (find)',
+      { requires: { auth: 'enabled' } },
+      async function () {
+        const findCommandToBeStarted = once(client, 'commandStarted');
+        await client.db().collection('test').findOne();
+        const [findCommandStarted] = await findCommandToBeStarted;
+
+        expect(findCommandStarted).to.have.property('commandName', 'find');
+        expect(client.options).to.not.have.property(Symbol.for('@@mdb.skipPingOnConnect'));
+        expect(client.s.options).to.not.have.property(Symbol.for('@@mdb.skipPingOnConnect'));
+
+        // Assertion is redundant but it shows that no initial ping is run
+        expect(findCommandStarted.commandName).to.not.equal('ping');
+
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+      }
+    );
+
+    it(
+      'automatically connects upon first operation (insertOne)',
+      { requires: { auth: 'enabled' } },
+      async function () {
+        const insertOneCommandToBeStarted = once(client, 'commandStarted');
+        await client.db().collection('test').insertOne({ a: 1 });
+        const [insertCommandStarted] = await insertOneCommandToBeStarted;
+
+        expect(insertCommandStarted).to.have.property('commandName', 'insert');
+        expect(client.options).to.not.have.property(Symbol.for('@@mdb.skipPingOnConnect'));
+        expect(client.s.options).to.not.have.property(Symbol.for('@@mdb.skipPingOnConnect'));
+
+        // Assertion is redundant but it shows that no initial ping is run
+        expect(insertCommandStarted.commandName).to.not.equal('ping');
+
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+      }
+    );
+
+    it(
+      'passes connection errors to the user through the first operation',
+      { requires: { auth: 'enabled' } },
+      async function () {
+        const client = this.configuration.newClient(
+          'mongodb://iLoveJavascript?serverSelectionTimeoutMS=100',
+          { monitorCommands: true }
+        );
+
+        const result = await client
+          .db('test')
+          .collection('test')
+          .findOne()
+          .catch(error => error);
+
+        expect(result).to.be.instanceOf(MongoServerSelectionError);
+        expect(client).to.be.instanceOf(MongoClient);
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+      }
+    );
+
+    it(
+      'client.close will not permit operations to auto reconnect',
```

Comment:
can we word this as an action though? e.g. "does not auto reconnect after client.close"